spec/services/encrypted_document_storage/document_writer_spec.rb

```diff
@@ -8,30 +8,53 @@
 
       result = EncryptedDocumentStorage::DocumentWriter.new.encrypt_and_write_document(
         front_image: front_image,
+        front_image_content_type: 'image/jpeg',
         back_image: back_image,
+        back_image_content_type: 'image/png',
       )
 
-      front_file_path = Rails.root.join('tmp', 'encrypted_doc_storage', result.front_uuid)
-      back_file_path = Rails.root.join('tmp', 'encrypted_doc_storage', result.back_uuid)
-      front_key = Base64.strict_decode64(result.front_encryption_key)
-      back_key = Base64.strict_decode64(result.back_encryption_key)
+      front_filename = Rails.root.join('tmp', 'encrypted_doc_storage', result.front_filename)
+      back_filename = Rails.root.join('tmp', 'encrypted_doc_storage', result.back_filename)
+      key = Base64.strict_decode64(result.encryption_key)
 
       aes_cipher = Encryption::AesCipher.new
 
-      written_front_image = aes_cipher.decrypt(
-        File.read(front_file_path),
-        front_key,
-      )
-      written_back_image = aes_cipher.decrypt(
-        File.read(back_file_path),
-        back_key,
-      )
+      written_front_image = aes_cipher.decrypt(File.read(front_filename), key)
+      written_back_image = aes_cipher.decrypt(File.read(back_filename), key)
 
       expect(written_front_image).to eq(front_image)
       expect(written_back_image).to eq(back_image)
     end
   end
 
+  describe '#build_filename_for_content_type' do
+    let(:filename) { described_class.new.build_filename_for_content_type(content_type) }
+    let(:content_type) { nil }
+
+    describe 'extension assigning' do
+      subject { File.extname(filename) }
+
+      context 'jpeg' do
+        let(:content_type) { 'image/jpeg' }
+        it { should eql('.jpeg') }
+      end
+
+      context 'png' do
+        let(:content_type) { 'image/png' }
+        it { should eql('.png') }
+      end
+
+      context 'nonsense' do
+        let(:content_type) { 'yabba/dabbadoo' }
+        it { should eql('') }
+      end
+
+      context nil do
+        it { should eql('') }
+      end
+    end
+  end
+
```

Comment:
I don't know if there's a better / more idiomatic way to do these sorts of small tests with lots of examples?

Reply:
I think `should`s kind of deprecated as of newer RSpec, I've seen `is_expected.to eq('')`